subsys/lorawan/lorawan.c

```diff
@@ -85,7 +85,10 @@ static void McpsConfirm(McpsConfirm_t *mcpsConfirm)
 	}
 
 	last_mcps_confirm_status = mcpsConfirm->Status;
-	k_sem_give(&mcps_confirm_sem);
+	/* mcps_confirm_sem is only blocked on in the MCPS_CONFIRMED case */
```

Comment:
Comment could be, ```Release semaphore only for the confirmed frames```

Reply:
I don't think the suggested comment is useful, that information can be obtain simply by reading the if statement.
The comment was added to describe why only some calls to `McpsConfirm` give the `mcps_confirm_sem` semaphore.
Having written that, it seems like the name would more appropriately be `confimed_msg_done_sem` or similar.